### ISSUE TITLE
macOS build broken (since Snippets)

### ISSUE THREAD

**Issue by weissi** [apple/swift-nio#2749]
```
/Users/johannes/devel/swift-nio/Snippets/NIOFileSystemTour.swift:89:37: error: 'valueForAttribute' is only available in macOS 10.15 or newer
87 |   // Where supported, the extended attributes of a file can also be accessed, read, and modified:
88 |   for attribute in try await directory.attributeNames() {
89 |     let value = try await directory.valueForAttribute(attribute)
   |                                     |- error: 'valueForAttribute' is only available in macOS 10.15 or newer
   |                                     `- note: add 'if #available' version check
90 |     print("Extended attribute '\(attribute)' has value '\(value)'")
91 |   }
[164/183] Linking NIOWebSocketClient
```

**Comment by weissi** [comment]
Since #2746 I think

**Comment by tayloraswift** [comment]
do we need to add `platforms: [.macOS(.v14)]` to the manifest?

**Comment by weissi** [comment]
> do we need to add `platforms: [.macOS(.v14)]` to the manifest?

No, that'd be a SemVer major break. It needs `if #available` or `@available(...)`

**Comment by Lukasa** [comment]
Yeah, we should add the appropriate availability guards to the snippet.

**Comment by Lukasa** [comment]
Resolved by #2750.
